applications/serial_lte_modem/src/slm_at_mqtt.c

```diff
@@ -377,8 +383,20 @@ static void client_init(void)
 	client.evt_cb = mqtt_evt_handler;
 	client.client_id.utf8 = ctx.cid;
 	client.client_id.size = strlen(ctx.cid);
-	client.password = NULL;
-	client.user_name = NULL;
+	if (strlen(ctx.uname) > 0) {
+		ctx.username.utf8 = ctx.uname;
+		ctx.username.size = strlen(ctx.uname);
+		client.user_name = &ctx.username;
+	} else {
+		client.user_name = NULL;
+	}
+	if (strlen(ctx.pword) > 0) {
+		ctx.password.utf8 = ctx.pword;
+		ctx.password.size = strlen(ctx.pword);
+		client.password = &ctx.password;
+	} else {
+		client.password = NULL;
+	}
```

Comment:
If there is no username, there should be no password, either.

Reply:
The username-password check is added into AT MQTT connect handler:

			if ((username_sz == 0) && (password_sz > 0)) {
				/* Password without username is invalid. */
				return -EINVAL;
			}

In the client initial phase, password is set only if the username exists:

	client.user_name = NULL;
	client.password = NULL;
	if (strlen(ctx.uname) > 0) {
		ctx.username.utf8 = ctx.uname;
		ctx.username.size = strlen(ctx.uname);
		client.user_name = &ctx.username;
		if (strlen(ctx.pword) > 0) {
			ctx.password.utf8 = ctx.pword;
			ctx.password.size = strlen(ctx.pword);
			client.password = &ctx.password;
		}
	}

Reply:
Thanks for the update